components/aws/sagemaker/common/_utils.py

```diff
@@ -471,11 +471,61 @@ def create_model_request(args):
 
     return request
 
+def endpoint_name_exists(client, endpoint_name):
+  endpoint_exists = False
+  if endpoint_name:
+      endpoints = client.list_endpoints(
+          MaxResults=99,
+          NameContains=endpoint_name,
+      )['Endpoints']
+      for i in range(len(endpoints)):
+          ep = endpoints[i]
+          if (ep['EndpointName'] == endpoint_name):
+            endpoint_exists = True
+            logging.info("Endpoint exists: " + ep['EndpointName'] + " - " + ep['EndpointArn'])
+  return endpoint_exists
+
+def endpoint_config_name_exists(client, endpoint_config_name):
+  ep_config_name_exists = False
+  if endpoint_config_name:
+      if client.describe_endpoint_config(
```

Comment:
If an endpoint doesn't exist, this will actually raise an exception rather than returning `False`. Your method of iterating over endpoints could be applied here, as well.

Your iteration and condition over all endpoints/configs could also be simplified to run within a single `any()` block:
```
configs = client.list_endpoint_configs()["EndpointConfigs"]
if any(map(lambda x: x["EndpointConfigName"]==endpoint_config_name, configs)):
  return True
```

The `map()` method here creates a list of booleans, that corresponding to whether the element at that index has an `EndpointConfigName` that matches `endpoint_config_name`. The `any()` then checks that any of those booleans are `True` i.e. did any endpoint config match the name we gave.

Reply:
Documentation doesn't list exception, but it appears we have seen it in the past. So, adding exception handling